### ISSUE TITLE
Doesn't change the value in the class.

### ISSUE THREAD

**Issue by JKwon0331** [ml-explore/mlx-examples#625]
Hi, I am not famlilar with Python, and I faced difficulty in implementing code using mlx.

Here is my code.
   
class RelPositionalEncoding(nn.Module):
    """
    Relative positional encoding module.
    Args:
        d_model: Embedding dimension.
        max_len: Maximum input length.
    """

    def __init__(self, d_model: int = 512, max_len: int = 5000):
        super(RelPositionalEncoding, self).__init__()
        self.d_model = d_model
        self.pe = None
        self.extend_pe(mx.zeros((1, max_len)))

    def extend_pe(self, x):
        if self.pe is not None:
            if self.pe.shape[1] >= x.shape[1] * 2 - 1:
                if self.pe.dtype != x.dtype:# or self.pe.device != x.device:
                    self.pe = self.pe.dtype(x.dtype)
                return

        pe_positive = mx.zeros((x.shape[1], self.d_model))
        pe_negative = mx.zeros((x.shape[1], self.d_model))
        position = mx.arange(0, x.shape[1], dtype=mx.float32)[:,mx.newaxis,]
        
        div_term = mx.exp(
            mx.arange(0, self.d_model, 2, dtype=mx.float32) * -(math.log(10000.0) / self.d_model)
        )
        pe_positive[:, 0::2] = mx.sin(position * div_term)
        pe_positive[:, 1::2] = mx.cos(position * div_term)
        pe_negative[:, 0::2] = mx.sin(-1 * position * div_term)
        pe_negative[:, 1::2] = mx.cos(-1 * position * div_term)
        
        pe_positive = pe_positive[mx.arange(pe_positive.shape[0],0,-1)-1,][mx.newaxis,]
        pe_negative = pe_negative[1:][mx.newaxis,]
        
        pe = mx.concatenate((pe_positive, pe_negative), axis=1)
              
        self.pe = pe
        

    def __call__(self, x):

        self.extend_pe(x)
        pos_emb = self.pe[
            :,
            self.pe.shape[1] // 2 - x.shape[1] + 1 : self.pe.shape[1] // 2 + x.shape[1],
        ]
        return pos_emb

I tried to transfer Pytorch code into mlx, but I have no idea why self.pe doesn't change.
It always has None, which is the initiated value in the init.

Could you let me know how to deal with this problem?

**Comment by awni** [comment]
This is at best a gotcha but probably a bug. There is a fix here https://github.com/ml-explore/mlx/pull/905

As a workaround, you can avoid setting `self.pe` in the `__init__` and then check `hasattr(self, "pe")` in the initialization method.
